src/components/PathwayExercise/ExerciseContent/Assessment/index.js

```diff
@@ -13,20 +13,101 @@ function Assessment({
   const [correct, setCorrect] = useState();
   const [solution, setSolution] = useState();
   const [submit, setSubmit] = useState();
   const [submitDisable, setSubmitDisable] = useState();
   const [status, setStatus] = useState();
   const [triedAgain, setTriedAgain] = useState(res?.attempt_count);
   const params = useParams();
+  const [type, setType] = useState("single");
+  const [wrongAnswer, setWrongAnswer] = useState();
 
   // Assessment submit handler
-  const submitAssessment = () => {
+  const isValuesCorrect = (value1, value2) => {
+    if (value1?.length !== value2?.length) {
+      return false; // Lengths are different, so they are not the same
+    }
+
+    for (let i = 0; i < value1?.length; i++) {
+      if (!value2?.includes(value1[i]?.value)) {
+        return false; // Elements are different, so they are not the same
+      }
+    }
+
+    return true; // Length and elements match, so they are the same
+  };
+
+  const isCorrect = isValuesCorrect(solution, answer);
+
+  const calculateSelections = () => {
+    let correctSelections = 0;
+    let incorrectSelections = 0;
+
+    answer &&
+      answer.forEach((option) => {
+        const isCorrect = solution?.some(
+          (correctOption) => correctOption?.value === option
+        );
+        if (isCorrect) {
+          correctSelections++;
+        } else {
+          incorrectSelections++;
+        }
+      });
+
+    return { correctSelections, incorrectSelections };
+  };
+
+  const { correctSelections, incorrectSelections } = calculateSelections();
+
+  // implementing this logic to check if the answer is partially correct or not
+
+  const [finalDesicion, setFinalDesicion] = useState("");
+
+  useEffect(() => {
+    if (answer?.length > 1) {
+      if (answer?.length == 3) {
+        if (correctSelections == 3 && incorrectSelections == 0) {
+          setFinalDesicion("correct");
+        } else if (correctSelections == 2 && incorrectSelections == 1) {
+          setFinalDesicion("partially incorrect");
+        } else if (correctSelections == 1 && incorrectSelections == 2) {
+          setFinalDesicion("partially correct");
+        }
+      } else if (answer.length == 2) {
+        if (correctSelections == 2 && incorrectSelections == 0) {
+          setFinalDesicion("correct");
+        } else if (correctSelections == 0 && incorrectSelections == 2) {
+          setFinalDesicion("incorrect");
+        } else if (correctSelections == 1 && incorrectSelections == 1) {
+          setFinalDesicion("partially correct");
+        }
+      }
+    } else {
+      if (solution?.length > 1) {
+        if (correctSelections == 1) {
+          setFinalDesicion("partially correct");
+        } else {
+          setFinalDesicion("incorrect");
+        }
+      } else {
+        if (answer?.[0] === solution?.[0]?.value) {
+          setFinalDesicion("correct");
+        } else {
+          setFinalDesicion("incorrect");
+        }
+      }
+    }
```

Comment:
This code should be reworked. Here are some of my concerns:
1. More code will have to be added if say the answer can have 4 correct answers in the future.
2. Too much repetition of strings (correct, incorrect, etc.) If these messages are changed in the future, they will have to be changed in multiple places.
3. Without seeing the code, it'll be difficult to know the difference between partially correct and partially incorrect, and it adds more code.
4. `===` should be used instead of `==`

Reply:
Hello @jschanker ,

I got your points.

1. For the first point, I am thinking of adding more code for the condition, if the answer.length === 4,  Please  Suggest me if any other better approach can be taken.

2. For the third point, I am unclear about this, do I need to change or add something to this?

3. Fixed point number 2 & 4.

Reply:
Hi @amansharmma, mujhe aur jaankari zarooree hai. "partially correct" aur "partially incorrect" mein kyaa farq hai? Back-end ka pratisaad se, mujhe lagta hai ki "partially correct" ka matlab kuch galat javaab chayan nahi kiya gaya. Lekin tumhara code se, "partially correct" ka matlab ek galat javaab ho sakta hai. Agar "partially correct" aur "partially incorrect" ke matlab back-end aata hai, yeh betar hoga hai.

![image](https://github.com/navgurukul/bhanwari-devi/assets/9159008/4d9186c1-2677-445e-9999-c802d246527d)

Reply:
@amansharmma Vistaar karna, humein pratisaad Android aur web samaan hona chahiye. Aur bhi, agar pratisaad ko badalnaa hai, toh humein web aur Android ke liye is code ko nahi badalna chahna hai. Shayad, back-end ek map sahee-galat jodon se pratissad ko bhej sakta hai. Udharan ke liye, vo`{"(3, 0)" => "correct", (2, 0) => "partially correct", etc.}` bhej saka hai. Phir, tum sirf lookup ke saath ek code ki line likhte hai.